my-fullstack-app-client/src/components/Products/Products.jsx

```diff
@@ -1,77 +1,83 @@
 
-export function Products({ minPrice, maxPrice }) {
+export function Products() {
   const [products, setProducts] = useState([]);
 
-  const { productCondition, productCategory, productOrderByPrice } =
-    useContext(FiltersContext);
+  const {
+    productCondition,
+    productCategory,
+    productOrderByPrice,
+    productTitle,
+  } = useContext(FiltersContext);
 
   const apiUrl = new URL("http://localhost:5000/products");
   productCategory && apiUrl.searchParams.append("category", productCategory);
-  minPrice && apiUrl.searchParams.append("minPrice", minPrice);
-  maxPrice && apiUrl.searchParams.append("maxPrice", maxPrice);
   productCondition && apiUrl.searchParams.append("condition", productCondition);
 
   useEffect(() => {
     fetch(apiUrl)
       .then((response) => response.json())
       .then((data) => {
-        console.log("condition", productCondition);
-        console.log("category", productCategory);
-        console.log("price", productOrderByPrice);
-        console.log("products", products);
         setProducts(data);
       })
       .catch((error) => {
         console.error("Error fetching products: ", error);
       });
     return () => {};
-  }, [
-    productCategory,
-    minPrice,
-    maxPrice,
-    productOrderByPrice,
-    productCondition,
-  ]);
+  }, [productCategory, productOrderByPrice, productCondition, productTitle]);
+
+  // if (productOrderByPrice === "low") {
+  //   products &&
+  //     products.sort(function (a, b) {
+  //       if (a.price > b.price) {
+  //         return 1;
+  //       }
+  //       if (a.price < b.price) {
+  //         return -1;
+  //       }
+  //       // a must be equal to b
+  //       return 0;
+  //     });
+  //   products.length = 10;
+  // }
+
+  // if (productOrderByPrice === "high") {
+  //   products &&
+  //     products.sort(function (a, b) {
+  //       if (a.price < b.price) {
+  //         return 1;
+  //       }
+  //       if (a.price > b.price) {
+  //         return -1;
+  //       }
+  //       // a must be equal to b
+  //       return 0;
+  //     });
+  //   products.length = 10;
+  // }
 
-  if (productOrderByPrice === "low") {
-    products.sort(function (a, b) {
-      if (a.price > b.price) {
-        return 1;
-      }
-      if (a.price < b.price) {
-        return -1;
-      }
-      // a must be equal to b
-      return 0;
-    });
-    products.length = 10;
-  }
+  orderByPrice(productOrderByPrice, products);
 
-  if (productOrderByPrice === "high") {
-    products.sort(function (a, b) {
-      if (a.price < b.price) {
-        return 1;
-      }
-      if (a.price > b.price) {
-        return -1;
-      }
-      // a must be equal to b
-      return 0;
-    });
-    products.length = 10;
-  }
+  const filteredProducts =
+    products.length > 0
+      ? products.filter((product) => {
+          return product.title
+            .toLowerCase()
+            .includes(productTitle.toLowerCase());
+        })
+      : [];
```

Comment:
Esto podria estar en useMemo
```suggestion
const filteredProducts = useMemo(() => {
  return products.filter((product) =>
    product.title.toLowerCase().includes(productTitle.toLowerCase())
  );
}, [products, productTitle]);
```